src/components/PivotTable/Droppable/Droppable.tsx

```diff
@@ -0,0 +1,166 @@
+  >
+
+  /**
+   * Array of keys currently present in this component
+   */
+  keyState: Array<keyof T>
+
+  /**
+   * Map of keys to the state of their filters
+   */
+  filterState?: Map<keyof T, Set<string>>
+
+  /**
+   * Function that updates the filterState of a key
+   */
+  handleFilterUpdate?: (key: keyof T, filtro: Set<string>) => void
+
+  /**
+   * Function that updates the keyState of this component
+   */
+  handleKeyUpdate: (values: Array<keyof T>) => void
+
+  /**
+   * Function used to navigate a draggable between droppables using the directional arrows
+   */
+  onKeyNav?: (dir: 'left' | 'right' | 'up' | 'down' | null, origin: string, key?: keyof T) => void
+}
+
+export interface DragItem<T> {
+  /**
+   * Name of the item, as an identifier
+   */
+  name: keyof T
+
+  /**
+   * The type of droppable item can be dropped on
+   */
+  type: string
+
+  /**
+   * Name of the origin droppable of the item
+   */
+  origin: string
+}
+
+export function Droppable<T>(props: DroppableProps<T>) {
+  const { name, keyState, keyMapping, type, filterState, keys, handleKeyUpdate, handleFilterUpdate, onKeyNav } = props
+
+  const keysWithFilters = Array.from(keys.keys()).filter((key) => keys.get(key).length > 0)
+
+  if (!filterState && keysWithFilters.length > 0) {
+    throw new Error(`The keys [${keysWithFilters}] has filters but the filterState prop is undefined`)
+  }
+
+  if ((filterState && !handleFilterUpdate) || (!filterState && handleFilterUpdate)) {
+    throw new Error('The filterState and handleFilterUpdate props must always be defined together')
+  }
+
+  const locale = useLocale()
+
+  const [{ isOver }, drag] = useDrop({
+    accept: type,
+    drop(item: DragItem<T>) {
+      if (!keyState.includes(item.name)) {
+        const newKeys = [...keyState, item.name]
+        handleKeyUpdate && handleKeyUpdate(newKeys)
+        return { result: true }
+      } else {
+        const pos = keyState.findIndex((element) => element === item.name)
+        const temp = [...keyState]
+        temp.splice(pos, 1)
+        const newKeys = [...temp, item.name]
+        handleKeyUpdate && handleKeyUpdate(newKeys)
+        return { result: false }
+      }
+    },
+    collect: (monitor) => ({
+      canDrop: !!monitor.canDrop(),
+      isOver: monitor.isOver() ? monitor.getItem().origin !== name : monitor.isOver(),
+    }),
+  })
+
+  function deleteByKey(id: keyof T) {
+    let tempKeys = [...keyState]
+    const index = tempKeys.indexOf(id)
+    tempKeys.splice(index, 1)
+    handleKeyUpdate && handleKeyUpdate(tempKeys)
+  }
+
+  const draggableButtons = keyState.map((key) => {
+    if (filterState && props.keys.get(key).length > 0) {
+      return (
+        <InternalFilterDraggable<T>
+          key={key as string}
+          type={type}
+          name={key}
+          filterItems={props.keys.get(key)}
+          selectedItems={props.filterState.get(key) || new Set<string>()}
```

Comment:
`new Set<string>()` only will be assigned to `selectedItems` if `props.filterState.get(key)` be `null`, right?

But if this happened, then in line 123 will throw an exception because of the  `props.keys.get(key).length`, right?

Or am I missing something?